jax/interpreters/xla.py

```diff
@@ -1305,9 +1305,13 @@ def _remat_translation_rule(c, axis_env, in_nodes,
   xb.parameter(dummy_subc, 0, c.get_shape(false_op), replicated=[])
 
   def zeros(xla_shape):
-    shape, dtype = xla_shape.dimensions(), xla_shape.numpy_dtype()
-    zero = xb.constant(dummy_subc, np.array(0, dtype=dtype))
-    return xops.Broadcast(zero, shape)
+    if xla_shape.is_array():
+      shape, dtype = xla_shape.dimensions(), xla_shape.numpy_dtype()
+      zero = xb.constant(dummy_subc, np.array(0, dtype=dtype))
+      return xops.Broadcast(zero, shape)
+    else:
```

Comment:
Maybe leave a TODO to plumb `Shape::IsToken` through xla.cc? Then we can write this as `elif xla_shape.is_token()` and have the `else` branch raise a TypeError.